controllers/contact-controller.js

```diff
@@ -270,6 +270,12 @@ const ContactController = Contact => {
   };
 
   const createContact = async (body, accountSid) => {
+    // if a contact has been already created with this taskId, just return it (idempotence on taskId). Should we use a different HTTP code status for this case?
+    if (body.taskId) {
```

Comment:
We would really like to use a transaction here: in the same transaction, we look for the record and then create it if not present.  That way we avoid a situation where two different processes check for the record at the same time, don't see it, and then both write it.

However, for this time, we can skip that.  It's highly unlikely in our current situation, and this is a clear improvement over what we have.

Reply:
Also, notice that if something ever goes wrong with a voice transfer (not a text transfer because they have different taskIds), and two different counselors end up with contact forms on the same task, whomever submits first wins.  I think that's fine (since it shouldn't happen anyway), but it's a good edge case to remember.